CRM/Contract/SepaLogic.php

```diff
@@ -582,7 +582,7 @@ public static function getMandateForRecurringContributionID($recurring_contribut
 
     // load mandate
     $mandate = civicrm_api3('SepaMandate', 'get', [
-      'entity_id'    => $recurring_contribution_id,
+      'id'    => $recurring_contribution_id,
```

Comment:
This doesn't seem right, the mandate ID is not the same as the recurring contribution ID it's referencing.